### PR TITLE
allow sample of first token on decode side

### PR DESCRIPTION
**Pull request by chenqianfzh**
This PR adds a new option to balance the workload of prefill & decode side when the system is under pressure. 

When the sglang of both prefill and decode are started with the option "--skip-sample-on-prefill True", the prefill side will not try to sample the first token. Instead, the logits of the first token will be sent over to decode side, which will do the sampling. Therefore, when the prefill side is under greater pressure, this option will help shift the load to the decode side a bit.

### CODE REVIEW

File: python/sglang/srt/disaggregation/utils.py [hunk 1]
```diff
@@ -96,6 +106,7 @@ def __init__(self, size: int, max_top_logprobs_num: int = 128):
         self.output_token_logprobs_idx = torch.zeros(
             (size, 16), dtype=torch.int32, device="cpu"
         )
+        # TODO: use cuda:0 if logits_only?
```
[the remark this answers is not in the document]
Comment (chenqianfzh) [reply python/sglang/srt/disaggregation/utils.py]: removed

File: python/sglang/srt/managers/scheduler.py [hunk 2]
```diff
@@ -658,18 +677,35 @@ def init_disaggregation(self):
...
-            self.disagg_metadata_buffers = MetadataBuffers(buffer_size)
+            if not self.skip_sample_on_prefill:
+                # *2 for the headroom.
+                buffer_size = self.max_running_requests * 2
+                self.req_to_metadata_buffer_idx_allocator = ReqToMetadataIdxAllocator(
+                    buffer_size
+                )
+                self.disagg_metadata_buffers = MetadataBuffers(buffer_size)
+            else:
+                assert self.model_config.vocab_size > 64
```
[the remark this answers is not in the document]
Comment (chenqianfzh) [reply python/sglang/srt/managers/scheduler.py]: actually, I don't need to worry about it, as the chance is next to zero. Even it happens, the connector can handle it, with minor perf downgrade.

File: python/sglang/srt/disaggregation/nixl/conn.py [hunk 3]
```diff
@@ -266,27 +276,28 @@ def add_transfer_request(
...
-                chunked_dst_kv_indice,
-                req.dst_gpu_id,
-                notif,
+                aux_index,
+                req.dst_aux_ptrs,
+                req.dst_aux_index,
+                str(req.room) + "_aux",
+                mem_type=mem_type,
             )
             handles.append(kv_xfer_handle)
```
[the remark this answers is not in the document]
Comment (chenqianfzh) [reply python/sglang/srt/disaggregation/nixl/conn.py]: thanks. fixed.
